account-model/src/account/mod.rs

```diff
@@ -0,0 +1,28 @@
+use std::convert::TryFrom;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum Classification {
+    PERSONAL,
+    BUSINESS,
+}
+
+impl Classification {
+    fn value(&self) -> &'static str {
+        match *self {
+            Classification::PERSONAL => "Personal",
+            Classification::BUSINESS => "Business",
+        }
+    }
+}
+
+impl TryFrom<&str> for Classification {
+    type Error = &'static str;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        match input {
+            "Personal" => Ok(Classification::PERSONAL),
+            "Business" => Ok(Classification::BUSINESS),
+            _ => Err("not recognised")
```

Comment:
rozumiem, że nijak nie jestem wstanie tutaj dorzucić wartości z inputu bo mam `&'static str` co rozumiem, że gdybym zaalokował `input + "not recognised"` to taka nowa zmienna będzie miała lifetime związany z tą funkcją. czy moge w takim razie coś zmienić w sygnaturze, żeby to zadziałało ?

Reply:
mozesz zwrocic po prostu `String`a a nie `&str`

Reply:
a jak to sie ma do lifenessu?

Reply:
lifetime'y sa (czasami) potrzebne tylko przy referencjach